perf/perf.py

```diff
@@ -0,0 +1,608 @@
+
+        writer_http_addr = 'localhost:%d' % (writer_id * 10000 + 8080)
+        writer_query_url = 'http://%s/iox/api/v1/databases/%s/query' % (writer_http_addr, db_name)
+        writer_query_params = {'q': 'select count(1) from sentinel'}
+
+        response = requests.get(url=writer_query_url, params=writer_query_params, timeout=10)
+        for i in range(20):
+            if response.ok:
+                break
+            print('waiting for round trip test to succeed')
+            time.sleep(0.5)
+            response = requests.get(url=writer_query_url, params=writer_query_params, timeout=10)
+
+        if not response.ok:
+            print(response.reason)
+            print(response.content)
+            return
+
+    else:
+        print()
+        print(writer_db_rules)
+        print()
+
+    print('created database "%s" on both IOx servers' % db_name)
+
+
+def run_test_battery(battery_name, router_id, writer_id, debug=False):
+    print('starting test battery "%s"' % battery_name)
+
+    # Write
+
+    battery_dir = os.path.join(os.getcwd(), battery_name)
+    datagen_filename = os.path.join(battery_dir, 'datagen.toml')
+    if debug:
+        iox_data_generator_path = os.path.abspath(os.path.join(os.getcwd(), '../target/debug/iox_data_generator'))
+    else:
+        iox_data_generator_path = os.path.abspath(os.path.join(os.getcwd(), '../target/release/iox_data_generator'))
+
+    router_http_addr = 'localhost:%d' % (router_id * 10000 + 8080)
+    args = [iox_data_generator_path,
+            '--host', router_http_addr, '--token', 'arbitrary',
+            '--org', org_name, '--bucket', bucket_name,
+            '--spec', datagen_filename]
+    env = {
+        'RUST_BACKTRACE': '0',
+    }
+    log_file = open('logs/test.log', mode='w')
+    if subprocess.run(args=args, stdout=log_file, stderr=log_file, env=env).returncode != 0:
+        raise ChildProcessError(
+            'failed to run iox_data_generator for battery "%s", check %s' % (battery_name, log_file.name))
+
+    # Query
+
+    writer_http_addr = 'localhost:%d' % (writer_id * 10000 + 8080)
+    query_url = 'http://%s/iox/api/v1/databases/%s/query' % (writer_http_addr, db_name)
+    queries_filename = os.path.join(battery_dir, 'queries.toml')
+    queries = toml.load(open(queries_filename))
+
+    for query in queries['queries']:
+        if 'sql' not in query:
+            print('query missing SQL query')
+            print(query)
+            print()
+            continue
+        sql = query['sql']
+        name = query['name']
+        if name is None:
+            name = sql
+
+        print('running test "%s"' % name)
+        time_start = time.time()
+        params = {'q': sql, 'format': 'csv'}
+        response = requests.get(url=query_url, params=params)
+        time_delta = '%dms' % math.floor((time.time() - time_start) * 1000)
+
+        if not response.ok:
+            print(response.reason)
+            print(response.content.decode('utf-8'))
```

Comment:
```suggestion
            print(response.text)
```

Same pattern is used in other places as well.